install/detect-platform.sh

```diff
@@ -0,0 +1,25 @@
+echo "${_group}Detecting Docker platform"
+
+
+# Sentry SaaS uses stock Yandex ClickHouse, but they don't provide images that
+# support ARM, which is relevant especially for Apple M1 laptops, Sentry's
+# standard developer environment. As a workaround, we use an altinity image
+# targeting ARM.
+#
+# See https://github.com/getsentry/self-hosted/issues/1385#issuecomment-1101824274
+#
+# Images built on ARM also need to be tagged to use linux/arm64 on Apple
+# silicon Macs to work around an issue where they are built for
+# linux/amd64 by default due to virtualization.
+# See https://github.com/docker/cli/issues/3286 for the Docker bug.
+
+if [[ "$(docker info --format '{{.Architecture}}')" = "x86_64" ]]; then
+    export DOCKER_PLATFORM="linux/amd64"
+    export CLICKHOUSE_IMAGE="yandex/clickhouse-server:20.3.9.70"
+else
```

Comment:
What do you think about improving these statements?
I mean currently `arm64` is being considered for every architecture and error which does not return `x86_64`.

I would say check for explicit `x86_64` and explicit `xxx` (I don't know what docker would output in arm) and if none of them was returned just drop and echo your system is not supported.

For example on a system which docker service is not started it returns:
```
$ docker info --format '{{.Architecture}}'

template: :1:2: executing "" at <.Architecture>: reflect: indirection through nil pointer to embedded struct field Info
```

Reply:
Yeah, this definitely could be better. I will make the changes you suggest, thanks!